saleor/checkout/calculations.py

```diff
@@ -355,22 +328,52 @@ def _calculate_and_add_tax(
     address: Optional["Address"] = None,
 ):
     if tax_calculation_strategy == TaxCalculationStrategy.TAX_APP:
-        # Call the tax plugins.
+        # If taxAppId is not configured remain previous behaviour to call plugin and apps
```

Comment:
Consider that "previous" only makes sense when you read this PR; it doesn't when you read the code later when it's merged. I would improve the comment to say that:
- `taxAppId` can refer to an app or a plugin (plugin ID is supported only for backward compatibility and will be removed)
-  when taxAppId is not provided, we run (all?) tax plugins and tax apps (what if both plugin is active, and apps are active?)
- when taxAppId is provided, run a plugin or app with this identifier.

Reply:
Good point.
About the question, the logic was like this before, I mean to run plugins then go Apps if they are all active, I'm not sure if that behaviour is expected, but probably shouldn't be changed lightly.
That's also one of the reason why `taxAppId` was introduced, as this is not what many customers may expect/want.

Reply:
Yes, this old behavior should not be changed and I just wasn't sure what were the assumptions here in the old code. But since we're adding more logic here I think it's worth describing in a comment.